scripts/stac/imagery/create_stac.py

```diff
@@ -40,15 +42,24 @@ def create_item(
     geometry, bbox = get_extents(gdalinfo_result)
 
     item = ImageryItem(id_, file, utc_now)
-    item.update_datetime(start_datetime, end_datetime)
-    item.update_spatial(geometry, bbox)
-    item.add_collection(collection_id)
 
     if derived_from is not None:
+        start_date_list = []
+        end_date_list = []
         for derived in derived_from:
+            derived_item_content = read(derived)
+            derived_stac = json.loads(derived_item_content.decode("UTF-8"))
+            start_date_list.append(derived_stac["properties"]["start_datetime"])
```

Comment:
Since we only need one of each of these, rather than adding all of them to a list and then computing the min/max, we can continuously replace the value if it's lower than the existing value:

```suggestion
        start_datetime = min(start_datetime, derived_stac["properties"]["start_datetime"])
```

You can compare RFC 3339 datetime strings without converting them to datetimes, so this should even be faster.